os/dnd.h

```diff
@@ -0,0 +1,106 @@
+// LAF OS Library
+// Copyright (C) 2024  Igara Studio S.A.
+//
+// This file is released under the terms of the MIT license.
+// Read LICENSE.txt for more information.
+
+#ifndef OS_DND_H_INCLUDED
+#define OS_DND_H_INCLUDED
+#pragma once
+
+#include "base/paths.h"
+#include "base/debug.h"
+#include "gfx/point.h"
+#include "os/surface.h"
+
+#include <memory>
+
+#pragma push_macro("None")
+#undef None // Undefine the X11 None macro
+
+namespace os {
+
+  class Window;
+
+  // Operations that can be supported by source and target windows in a drag
+  // and drop operation.
+  enum DropOperation {
+    None = 0,
+    Copy = 1,
+    Move = 2,
+    Link = 4,
+    Any = Copy | Move | Link,
+  };
+
+  // Types of representations supported for each DragDataItem.
+  enum DragDataItemType {
+    Paths,
+    Image
+  };
+
+  // Interface to get dragged data from the platform's implementation.
+  class DragDataProvider {
+  public:
+    virtual ~DragDataProvider() {}
+    virtual base::paths getPaths() = 0;
+    virtual std::unique_ptr<os::Surface> getImage() = 0;
```

Comment:
We can use the `SurfaceRef` directly here because `Surface` is a `RefCount` object (has a reference counter included):
```suggestion
    virtual os::SurfaceRef getImage() = 0;
```